chemprop/featurizers/atom.py

```diff
@@ -30,50 +32,17 @@ class MultiHotAtomFeaturizer(AtomFeaturizer):
     * hybridization
     * aromaticity
     * mass
-
-    The feature vectors produced by this featurizer have the following (general) signature:
-
-    +---------------------+-----------------+--------------+
-    | slice [start, stop) | subfeature      | unknown pad? |
-    +=====================+=================+==============+
-    | 0-101               | atomic number   | Y            |
-    +---------------------+-----------------+--------------+
-    | 101-108             | degree          | Y            |
-    +---------------------+-----------------+--------------+
-    | 108-114             | formal charge   | Y            |
-    +---------------------+-----------------+--------------+
-    | 114-119             | chiral tag      | Y            |
-    +---------------------+-----------------+--------------+
-    | 119-125             | # Hs            | Y            |
-    +---------------------+-----------------+--------------+
-    | 125-131             | hybridization   | Y            |
-    +---------------------+-----------------+--------------+
-    | 131-132             | aromatic?       | N            |
-    +---------------------+-----------------+--------------+
-    | 132-133             | mass            | N            |
-    +---------------------+-----------------+--------------+
-
-    NOTE: the above signature only applies for the default arguments, as the each slice (save for
-    the final two) can increase in size depending on the input arguments.
     """
-
-    max_atomic_num: InitVar[int] = 100
-    degrees: Sequence[int] = field(default_factory=lambda: range(6))
-    formal_charges: Sequence[int] = field(default_factory=lambda: [-1, -2, 1, 2, 0])
-    chiral_tags: Sequence[int] = field(default_factory=lambda: range(4))
-    num_Hs: Sequence[int] = field(default_factory=lambda: range(5))
-    hybridizations: Sequence[HybridizationType] = field(
-        default_factory=lambda: [
-            HybridizationType.SP,
-            HybridizationType.SP2,
-            HybridizationType.SP3,
-            HybridizationType.SP3D,
-            HybridizationType.SP3D2,
-        ]
-    )
-
-    def __post_init__(self, max_atomic_num: int = 100):
-        self.atomic_nums = {i: i for i in range(max_atomic_num)}
+    
+    atomic_nums: Sequence[int] = field(default_factory=list)
+    degrees: Sequence[int] = field(default_factory=list)
+    formal_charges: Sequence[int] = field(default_factory=list)
+    chiral_tags: Sequence[int] = field(default_factory=list)
+    num_Hs: Sequence[int] = field(default_factory=list)
+    hybridizations: Sequence[HybridizationType] = field(default_factory=list)
```

Comment:
Can you elaborate a bit more? We use list here because it is easier for user to specify these attributes as list. For example: formal_charges = [-1, -2, 1, 2, 0]. Subsequently in the post_init function we convert this into a dictionary because we use the dictionary to build multi-hot encoding. For example: self.formal_charges = {j: i for i, j in enumerate(self.formal_charges)}, where we re-indexed the formal charges provided by user.

I understand that we are not consistent with the datatype for the attributes since we expect an input of list but internally convert it to dictionary. I suggest to rename the variables in post init with an underscore (e.g., self._formal_charges = ...) to resolve the confusion. What do you think?

Reply:
the current setup allows you to init a `MultiHotAtomFeaturizer()` with 0 choices for each subfeature. This makes no sense and should not even be allowed. At this point, the class should probably not even be a `dataclass` because we do so much transformation to the input arguments and rely heavily on `__post_init__()` to make the class work. We should just make it into a traditional class and define a `__init__()`